### DISCUSSION TITLE
Active Disk Monitoring OFF (Bug?)

### DISCUSSION THREAD

**Discussion by gjrtimmer** [minio/minio#18909] [Q&A] [answered]
So because of my storage backend, I'm trying to turn of the active Disk monitoring.
But I'm not able to do so with the env_var `_MINIO_DRIVE_ACTIVE_MONITORING`

I've looked at the code.([Link-2-Source](https://github.com/minio/minio/blob/2ddf2ca9344ead19d91b6679ff2a1073724fa273/cmd/xl-storage-disk-id-check.go#L837))

```go
// diskActiveMonitoring indicates if we have enabled "active" disk monitoring
var diskActiveMonitoring = true

// Indicates if users want to enable drive_quorum feature
var driveQuorum bool

func init() {
	diskActiveMonitoring = (env.Get("_MINIO_DRIVE_ACTIVE_MONITORING", config.EnableOn) == config.EnableOn) ||
		(env.Get("_MINIO_DISK_ACTIVE_MONITORING", config.EnableOn) == config.EnableOn)
```

But this always evaluates to `true` right? Might this be a bug that avoids me to turning off active disk monitoring?

EDIT --- Overlooked 2 different env_vars

Please advise? Is this a bug?

**Comment by harshavardhana** [comment]
it is a bug, however you can turn-it off via `_MINIO_DRIVE_ACTIVE_MONITORING=off` and also `_MINIO_DISK_ACTIVE_MONITORING=off`

**Reply by gjrtimmer** [reply]
@harshavardhana I tried turning it off, but the boolean expression always evaluates to True, so it will not turn off.

**Reply by gjrtimmer** [reply]
You cannot turn it off with the mentioned variables, I will research why not.

**Comment by gjrtimmer** [comment]
@harshavardhana I think I found the problem, why it will not turn off, I will submit a PR with Proof

**Reply by harshavardhana** [reply, marked as answer]
Don't bother I have sent a fix which is needed for a different reason which includes this as well.

**Reply by gjrtimmer** [reply]
@harshavardhana Thank you I just did some testing with the code, and everything works well when you change the expression from `||` to `&&`

**Reply by gjrtimmer** [reply]
Thank you. Hopefully, it will get merged soon.

**Comment by gjrtimmer** [comment]
@harshavardhana does this fix also fixes other stuff related to this.
I tried with my own build, disk monitoring is off but I still get the same error.

```bash
[Servers](https://nomad.tpi.local/ui/servers)
[Topology](https://nomad.tpi.local/ui/topology)
OPERATIONS

[Evaluations](https://nomad.tpi.local/ui/evaluations)
[Access Control](https://nomad.tpi.local/ui/access-control)
v1.7.3
 
Word Wrap

   
_MINIO_DRIVE_ACTIVE_MONITORING=off
_MINIO_DISK_ACTIVE_MONITORING=on
diskActiveMonitoring=false
MinIO Object Storage Server
Copyright: 2015-0000 MinIO, Inc.
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Version: DEVELOPMENT.GOGET (go1.21.6 linux/arm64)

Status:         1 Online, 0 Offline. 
S3-API: http://172.17.0.17:9000  http://127.0.0.1:9000 
Console: https://minio.tpi.local 

Documentation: https://min.io/docs/minio/linux/index.html
Warning: The standard parity is set to 0. This can lead to data loss.

API: SYSTEM()
Time: 04:58:08 UTC 01/30/2024
DeploymentID: a45bab1f-4190-4877-a2ee-3334e7530835
Error: node(127.0.0.1:9000): taking drive /data offline, time since last response 30.112s (*errors.errorString)
       7: /Users/gtimmer/Work/minio/internal/logger/logger.go:249:logger.LogAlwaysIf()
       6: /Users/gtimmer/Work/minio/cmd/xl-storage-disk-id-check.go:1019:cmd.(*xlStorageDiskIDCheck).checkHealth()
       5: /Users/gtimmer/Work/minio/cmd/xl-storage-disk-id-check.go:978:cmd.(*xlStorageDiskIDCheck).waitForToken()
       4: /Users/gtimmer/Work/minio/cmd/xl-storage-disk-id-check.go:942:cmd.(*xlStorageDiskIDCheck).TrackDiskHealth()
       3: /Users/gtimmer/Work/minio/cmd/xl-storage-disk-id-check.go:496:cmd.(*xlStorageDiskIDCheck).ReadFileStream()
       2: /Users/gtimmer/Work/minio/cmd/bitrot-streaming.go:172:cmd.(*streamingBitrotReader).ReadAt()
       1: /Users/gtimmer/Work/minio/cmd/erasure-decode.go:168:cmd.(*parallelReader).Read.func1()
```

Any suggestions?

**Reply by klauspost** [reply]
This is unrelated to the env vars. Your disks are completely unresponsive and there is a huge queue built up. MinIO cannot operate in such an environment - so it very reasonably takes the disk offline. This is what you want.

Your options are to A) Reduce load or B) Provide MinIO with more appropriate storage.
